check_exec.go

```diff
@@ -37,28 +38,45 @@ func (c CheckExec) Execute(timeout time.Duration) error {
 
 	p, err := os.StartProcess(path, fullArgs, &os.ProcAttr{})
 	if err != nil {
-		return err
+		return false, err
 	}
 
-	done := make(chan error, 1)
+	// assume the exit codes of the process mean 0 => true, 1 => false, 2 => error
+	done := make(chan bool, 1)
+	fail := make(chan error, 1)
 	go func() {
 		processState, err := p.Wait()
 		if err != nil {
-			done <- err
+			fail <- err
 		} else if processState.Success() {
-			done <- nil
+			done <- true
 		} else {
-			done <- fmt.Errorf("Non-zero exit code")
+			status, ok := processState.Sys().(syscall.WaitStatus)
```

Comment:
More often than not, you'll see this turned into a single statement like `if status, ok := processState.Sys().(syscall.WaitStatus); ok {` instead of two lines. It just helps with scope containment.